### PR TITLE
Add `wp post meta clean-duplicates <id> <key>` command

### PR DESCRIPTION
**Pull request by BhargavBhandari90**
Fixes: #356 

Need suggestions/corrections for labelling. 

QUESTION: Should we do the same for the user and term meta commands.?

### CODE REVIEW

File: features/post-meta.feature [hunk 1]
```diff
@@ -488,3 +488,46 @@ Feature: Manage post custom fields
...
+      y
+      """
+
+    When I run `wp post meta add 1 foo bar`
+    Then STDOUT should be:
+      """
+      Success: Added custom field.
+      """
+
+    When I try the previous command again
```
Comment (danielbachhuber) [inline features/post-meta.feature:509]:
```suggestion
    When I run the previous command again
```

File: features/post-meta.feature [hunk 2]
```diff
@@ -488,3 +488,46 @@ Feature: Manage post custom fields
...
+    Then the return code should be 0
+
+    When I try the previous command again
```
Comment (danielbachhuber) [inline features/post-meta.feature:512]:
```suggestion
    When I run the previous command again
```

File: features/post-meta.feature [hunk 3]
```diff
@@ -488,3 +488,46 @@ Feature: Manage post custom fields
...
+    Then the return code should be 0
+
```
Comment (danielbachhuber) [inline features/post-meta.feature:514]: Can we add a test here to verify there are three `foo` meta keys?

File: features/post-meta.feature [hunk 4]
```diff
@@ -488,3 +488,46 @@ Feature: Manage post custom fields
...
+    When I run `wp post meta clean-duplicates 1 foo < session_no`
+    # Check for contains only, as the string contains a trailing space.
+    Then STDOUT should contain:
+      """
+      Are you sure you want to delete 2 duplicate enclosures and keep 1 valid enclosures? [y/n]
+      """
+
```
Comment (danielbachhuber) [inline features/post-meta.feature:521]: Can we add an assertion here to verify there are still three meta keys?

File: features/post-meta.feature [hunk 5]
```diff
@@ -488,3 +488,46 @@ Feature: Manage post custom fields
...
+    When I run `wp post meta clean-duplicates 1 foo < session_yes`
+    Then STDOUT should contain:
+      """
+      Success: Cleaned up duplicate enclosures.
+      """
+
```
Comment (danielbachhuber) [inline features/post-meta.feature:527]: Can we run `wp post meta clean-duplicates 1 foo < session_yes` again, and verify "Nothing to clean up:" is displayed?

File: src/Post_Meta_Command.php [hunk 6]
```diff
@@ -97,18 +97,86 @@ protected function get_metadata( $object_id, $meta_key = '', $single = false ) {
...
+	 *
+	 * <id>
+	 * : ID of the post to clean.
+	 *
+	 * <key>
+	 * : Meta key to clean up.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     wp post meta clean-duplicates 1234 enclosure
```
Comment (danielbachhuber) [inline src/Post_Meta_Command.php:128]: Can we add a comment above this example explaining what it does?

**Review by danielbachhuber — CHANGES_REQUESTED** [review]
Great start! Left a few comments on the test conditions.

> Need suggestions/corrections for labelling.

Not sure I follow... can you explain further?

> Should we do the same for the user and term meta commands.?

I think we can hold off for now. Should an actual need arise in the future, it should be simple enough to abstract this out.

File: features/post-meta.feature (hunk 4 above)
Comment (BhargavBhandari90) [reply features/post-meta.feature:528]: Sorry, I didn't get this.

**Comment by BhargavBhandari90** [comment]
@danielbachhuber 


>Need suggestions/corrections for labelling.

>> Not sure I follow... can you explain further?

I mean command is `clean-duplicates` and we are displaying a success message like `Cleaned up duplicate enclosures.`

So the word `enclosures` is correct? OR should we have something like `Cleaned up duplicate metadat.`?

**Comment by danielbachhuber** [comment]
> I mean command is `clean-duplicates` and we are displaying a success message like `Cleaned up duplicate enclosures.`
> 
> So the word `enclosures` is correct? OR should we have something like `Cleaned up duplicate metadata.`?

@BhargavBhandari90 Oh, I see.

The message should be something like `Cleaned up duplicate '<meta-key>' meta values.`

File: features/post-meta.feature (hunk 4 above)
Comment (danielbachhuber) [reply features/post-meta.feature:528]: Something like:

```
When I run `wp post meta list 1 --keys=foo --format=count`
Then STDOUT should be:
  """
  3
  """
```

File: features/post-meta.feature (hunk 2 above)
Comment (BhargavBhandari90) [reply features/post-meta.feature]: Done.

File: features/post-meta.feature (hunk 4 above)
Comment (BhargavBhandari90) [reply features/post-meta.feature:534]: Done.

File: features/post-meta.feature (hunk 5 above)
Comment (BhargavBhandari90) [reply features/post-meta.feature:546]: Done.

File: src/Post_Meta_Command.php (hunk 6 above)
Comment (BhargavBhandari90) [reply src/Post_Meta_Command.php:129]: Done.

File: src/Post_Meta_Command.php [hunk 7]
```diff
@@ -97,18 +97,88 @@ protected function get_metadata( $object_id, $meta_key = '', $single = false ) {
...
+	 *
+	 * <id>
+	 * : ID of the post to clean.
+	 *
+	 * <key>
+	 * : Meta key to clean up.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     # Delete dulpicate post meta.
```
Comment (danielbachhuber) [inline src/Post_Meta_Command.php:128]:
```suggestion
	 *     # Delete duplicate post meta.
```

**Review by danielbachhuber — APPROVED** [review]
Great work on this, @BhargavBhandari90 ! 🎉 

I made a few more changes you might find useful to review ^

Overall, you've did a great job 😊

**Comment by BhargavBhandari90** [comment]
Thank You @danielbachhuber 

>I made a few more changes you might find useful to review ^

Yep, I saw that.

**Comment by schlessera** [comment]
Great work on this!

A minor thing: This PR messed up the formatting of the JSON files in the Behat tests. I'll get that cleaned up again in a subsequent PR.

**Pull request merged**